include/ignition/gui/qml/Main.qml

```diff
@@ -325,24 +344,59 @@ ApplicationWindow
     }
   }
 
+  Timer {
+    id: timer
+  }
+
   /**
    *  Confirmation dialog on close button
    */
   Dialog {
     id: confirmationDialogOnExit
-    title: "Do you really want to exit?"
+    title: (dialogOnExitText ? dialogOnExitText : "Do you really want to exit?")
+    objectName: "confirmationDialogOnExit"
 
     modal: true
     focus: true
     parent: ApplicationWindow.overlay
-    width: 300
+    width: 500
     x: (parent.width - width) / 2
     y: (parent.height - height) / 2
     closePolicy: Popup.CloseOnEscape
-    standardButtons: Dialog.Ok | Dialog.Cancel
-
-    onAccepted: {
-      Qt.quit()
+    standardButtons:
+      (exitDialogShowCloseGui ? Dialog.Ok : Dialog.NoButton) |
+      (exitDialogShowShutdown ? Dialog.Discard : Dialog.NoButton) |
+      Dialog.Cancel
+
+    // The button texts need to be changed later than in onCompleted as standardButtons change later
+    onAboutToShow: function () {
+      if (exitDialogShowCloseGui)
+        footer.standardButton(Dialog.Ok).text =
+          (exitDialogCloseGuiText ? exitDialogCloseGuiText : "Close GUI")
```

Comment:
Just a final nitpick, I was using `ign-gazebo` with this PR and noticed that it's a bit confusing to see this by default. Can we default to `Close`? If you're planning to follow up with an `ign-gazebo` PR this is less concerning, but I think that in general "GUI" can be a strange word to display to users by default for other applications too.

![image](https://user-images.githubusercontent.com/5751272/149590134-70fa4734-0a71-4009-a080-92e335865c02.png)

Reply:
I understand your request to change the text, but imagine a dialog with "Cancel" and "Close". I hate these dialogs. Does "Close" close just the dialog, or the whole app? Before this PR, there was just "OK", which was good when there were just two options, but would be weird with three.

I could make the logic a bit more complicated and test whether there are just two, or all three buttons displayed. If only two were displayed, the text of the non-cancel button would be just "OK" (regardless if it is close gui or shutdown button). And if all three would be displayed, then the texts of the buttons could be as they are now. And in case any of the texts is customized, it would be used in all cases.

Another option would be to leave it completely on the users to redefine the button texts whenever they display all 3 buttons. But if only the shutdown button text were defined, it would lead to a dialog with "Shutdown server and GUI", "Cancel" and "OK" buttons, which is not best...

Reply:
> I hate these dialogs. 

Ah yeah good point, that's horrible.

> Another option would be to leave it completely on the users to redefine the button texts whenever they display all 3 buttons.

I think that's reasonable, if the user is customizing the dialog, they should be in a position to edit all buttons.

> "OK" (regardless if it is close gui or shutdown button)

Along the lines of the comment above, I think that's ok. When the user starts customizing the dialog, when may change the prompt and button texts as needed.

Reply:
Good, I implemented your latest suggestion. I also added a few notes to docs and the tutorial, which should be enough to inform users of this API about the need of changing the default text value.